### PR TITLE
Fixes #97 - Internal Links and TOC updates

### LINKED ISSUE #97: Internal Links are not working
Issue:

Links to other sections of the wiki (not deep-linked) are not clickable in the generated PDF

image

**To Reproduce:**

Add a link to a wiki MD page:

~~~markdown
I want to test if this [link](/Admin-Layout-and-Customization) works
~~~

Run the build (ctrl+f5)

**Expected**

Link to internal section of the PDF will work.

**Actual**

The text is styled as a link but does not work.

**Example**

Added the above to the test wiki in the repo and built the project. This was output in the console:

~~~html
<div style='page-break-after: always;'><h1>Another Page</h1><a id="another-page">&nbsp;</a><h2 id="another-page-here">Another Page Here</h2>
<p>I want to test if this <a href="/Admin-Layout-and-Customization">link</a> works</p>
</div>
  html:
<h1>Admin Layout and Customization</h1><a id="admin-layout-and-customization">&nbsp;</a><h2 id="smileys">Smileys</h2>
~~~

Attached PDF the link does not work (I took out the images at the top of the page for clarity in the console

[export.pdf](https://github.com/MaxMelcher/AzureDevOps.WikiPDFExport/files/8109837/export.pdf)

### PR DESCRIPTION
**Pull request by Diom**
TOC:
* Add a class to allow for page breaking
* Fixes for ~ fenced code blocks to be supported

Internal Links:
* Detects the base folder by finding the .attachements directory
* Allows for "code-wikis" to work with internal links
* Fixes minor issue with ':' character not working because of inconsistent URL escaping

### CODE REVIEW

File: AzureDevOps.WikiPDFExport/WikiPDFExporter.cs [hunk 1]
```diff
@@ -638,7 +631,7 @@ internal List<string> CreateGlobalTableOfContent(List<string> contents)
 
             foreach (var content in filteredContentList) 
             {
-                var headerMatches = Regex.Matches(content, "^ *#+ ?.*$", RegexOptions.Multiline);
+                var headerMatches = Regex.Matches(content, "^ *#{1,2} ?[^#].*$", RegexOptions.Multiline);
```
Comment (Diom) [inline AzureDevOps.WikiPDFExport/WikiPDFExporter.cs:634]:
```suggestion
                var headerMatches = Regex.Matches(content, "^ *#+ ?[^#].*$", RegexOptions.Multiline);
```
Revert the limit on the number of headings... was limiting my own doc. Should make that configurable in future.

File: AzureDevOps.WikiPDFExport/WikiPDFExporter.cs [hunk 2]
```diff
@@ -950,6 +949,17 @@ async private Task<string> generateWorkItemLink(string stringId, WorkItemTrackin
...
+        private String FindBaseOfWikiFromChildPath(String exportBase){
+            var directory = new DirectoryInfo(Path.GetFullPath(exportBase));
+            DirectoryInfo[] gitDirs = directory.GetDirectories("./.git");
+            if (gitDirs == null || gitDirs.Length == 0) {
+
+            }
+            Log($"Reading .order file in directory {exportBase}");
+            var orderFiles = directory.GetFiles(".order", SearchOption.TopDirectoryOnly);
+            return Path.GetFullPath("ds");
+        }
```
Comment (Diom) [inline AzureDevOps.WikiPDFExport/WikiPDFExporter.cs:962]: Need to remove this

**Comment by Diom** [comment]
@MaxMelcher Hey Max, wasn't sure about modifying the "code" example. Let me know if you have any concerns. I can add it a new wiki folder instead if you would prefer.

File: AzureDevOps.WikiPDFExport/WikiPDFExporter.cs [hunk 3]
```diff
@@ -769,10 +762,16 @@ public void CorrectLinksAndImages(MarkdownObject document, FileInfo file, Markdo
...
                         else if (link.Url.StartsWith("/"))
                         {
-                            //urls could be encoded and contain spaces - they are then not found on disk
-                            var linkUrl = HttpUtility.UrlDecode(link.Url);
-                            linkUrl = linkUrl.Replace("#", "-");
-                            absPath = Path.GetFullPath(_rootWikiPath + linkUrl);
+                            // Add URL replacements here
+                            var replacements = new Dictionary<string, string>(){
+                                {":", "%3A"},
+                                {"#", "-"}
```
Comment (MaxMelcher) [inline AzureDevOps.WikiPDFExport/WikiPDFExporter.cs:768]: %20 was missing - that happens if you have a space in your filename/picture.

**Comment by MaxMelcher** [comment]
> @MaxMelcher Hey Max, wasn't sure about modifying the "code" example. Let me know if you have any concerns. I can add it a new wiki folder instead if you would prefer.

no worries. Adding stuff here increases the test coverage :)

**Pull request merged**